src/Confluent.Kafka.Avro/AvroProducer.cs

```diff
@@ -0,0 +1,349 @@
+// Copyright 2018 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Derived from: rdkafka-dotnet, licensed under the 2-clause BSD License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Confluent.SchemaRegistry;
+using Confluent.Kafka.AvroSerdes;
+using Avro.Generic;
+
+
+namespace Confluent.Kafka.AvroSerdes
+{
+    /// <summary>
+    ///     Extends <see cref="Confluent.Kafka.Producer" /> with functionality
+    ///     for managing <see cref="Confluent.Kafka.AvroSerdes.AvroSerializer{T}" />
+    ///     instances.
+    /// </summary>
+    public class AvroProducer : Producer
+    {
+        private ISchemaRegistryClient schemaRegistryClient;
+
+        private Dictionary<Type, object> avroSerializers
+            = new Dictionary<Type, object>();
+
+        /// <summary>
+        ///     Creates a new AvroProducer instance.
+        /// </summary>
+        /// <param name="schemaRegistryClient">
+        ///     An implementation of ISchemaRegistryClient used for
+        ///     communication with Confluent Schema Registry.
+        /// </param>
+        /// <param name="config">
+        ///     A collection of librdkafka configuration parameters.
+        /// </param>
+        public AvroProducer(ISchemaRegistryClient schemaRegistryClient, IEnumerable<KeyValuePair<string, string>> config)
+            : base(config)
+        {
+            this.schemaRegistryClient = schemaRegistryClient;
+        }
+
+        /// <summary>
```

Comment:
:+1:

Reply:
there are edge cases where you would need to create more than one producer, but I think those are unlikely / uncommon, and the type lookup is overall a good solution. for example if you wanted different naming strategies for different topics with the same schema family (currently not supported by .net, but we'll add it) AND you wanted to produce to the topics from the same application you'd need two different serializers associated with the same type so you'd need two producers (or to explicitly register the serializer immediately before each produce). 

if you have a specific scenario of concern, please elaborate.

Reply:
no, I got rid of the serializing producer concept and also the ability to construct producers from an underlying `Handle` (can only do that with `AdminClient). The latter could be brought back if there was demand, but I don't think there will be.